hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -879,9 +880,8 @@ private boolean checkUserError(int responseStatusCode) {
    * @return boolean whether exception is due to MD5Mismatch or not
    */
   protected boolean isMd5ChecksumError(final AzureBlobFileSystemException e) {
-    return ((AbfsRestOperationException) e).getStatusCode()
-        == HttpURLConnection.HTTP_BAD_REQUEST
-        && e.getMessage().contains(MD5_ERROR_SERVER_MESSAGE);
+    AzureServiceErrorCode storageErrorCode = ((AbfsRestOperationException) e).getErrorCode();
```

Comment:
check if it can be case`e` is instanceof AbfsRestOperationException.

Reply:
see my comment above about making the catch on L787 a AbfsRestOperationException

Reply:
In line with steve's suggestion here.